### DISCUSSION TITLE
[RRFC]  turn off file name scrubbing for your own private registry

### DISCUSSION THREAD

**Discussion by claudiahdz** [npm/rfcs#65] [General]
Suggested in https://github.com/npm/cli/pull/251

**Comment by haakemon** [comment]
So whats the discussion needed around this item? I was eagerly awaiting the linked PR, because scrubbing breaks audit run in certain cases f.ex as described here (https://npm.community/t/npm-audit-returns-400-from-registry-when-non-registry-packages-satisfy-specs-that-exist-in-the-registry/462 ), and honestly I dont really need any scrubbing features. I would prefer just a simple flag to turn it off globally.

**Comment by jweinsteincbt** [comment]
Related: https://github.com/npm/npm/issues/20604

**Comment by darcyclarke** [comment]
@haakemon Appreciate the interest! This issue was created based on @isaacs' feedback here:

>This should be a config value with docs and tests. I'd prefer not to bury ad hoc environment variable switches in the cli's command definitions, or it gets unmaintainable real fast.

Our hope is that we can move the discussion here to round out a more succinct implementation for something like `--no-scrub`. 

I know @jweinsteincbt had some suggestions/questions that we can carry over as well:

>  Should this be limited to members of the org for that scope? Would need to cache the credentials so it does not need to do multiple lookups. Might not work for all registries (npm caching and so on). Also, that makes it harder to decide not to not require modules which have bad audit results if they are @ scoped.

We're definitely open to talking about this even at our [Open RFC call today at 2:00pm ET](https://github.com/npm/rfcs/issues/66) if you'd like to jump on to flush out some of the details.

**Comment by haakemon** [comment]
Sorry, I did not have a chance to join the call.

What was the outcome of this? I read through the notes, but could not make out any decision (if any).

Thanks.

**Comment by IgorCreator** [comment]
Hey guys
The issue https://github.com/npm/npm/issues/20604# have been archived already.

So I'll live comment here.
Next Solution might work:

Turning off npm audit on package installation
To turn off npm audit when installing a single package, use the --no-audit flag:
> npm install example-package-name --no-audit

Installing all packages
To turn off npm audit when installing all packages, set the audit setting to false in your user and global .npmrc config files or:
> npm set audit false
